packages/shared/src/components/auth/RegistrationForm.tsx

```diff
@@ -81,49 +81,64 @@ const RegistrationForm = ({
 }: RegistrationFormProps): ReactElement => {
   const { email } = useAuthData();
   const router = useRouter();
-  const { logEvent } = useContext(LogContext);
+  const { logEvent } = useLogContext();
   const [turnstileError, setTurnstileError] = useState<boolean>(false);
+  const [turnstileLoaded, setTurnstileLoaded] = useState<boolean>(false);
+  const [turnstileErrorLoading, setTurnstileErrorLoading] =
+    useState<boolean>(false);
   const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
   const [name, setName] = useState('');
   const isAuthorOnboarding = trigger === AuthTriggers.Author;
   const { username, setUsername } = useGenerateUsername(name);
-  const ref = useRef<TurnstileInstance>(null);
+  const turnstileRef = useRef<TurnstileInstance>(null);
   const isReorderExperiment = useFeature(featureOnboardingReorder);
   const isOnboardingExperiment = !!(
     router.pathname?.startsWith('/onboarding') && isReorderExperiment
   );
 
+  const logRef = useRef<typeof logEvent>();
+  logRef.current = logEvent;
+
   useEffect(() => {
-    logEvent({
+    logRef.current({
       event_name: AuthEventNames.StartSignUpForm,
     });
-    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
-    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   useEffect(() => {
     if (Object.keys(hints).length) {
-      logEvent({
+      logRef.current({
         event_name: AuthEventNames.SubmitSignUpFormError,
         extra: JSON.stringify({ error: hints }),
       });
       if (hints?.csrf_token) {
         setTurnstileError(true);
       }
-      ref?.current?.reset();
+      turnstileRef?.current?.reset();
     }
-    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
-    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [hints]);
 
+  useEffect(() => {
+    const turnstileLoadTimeout = setTimeout(() => {
```

Comment:
Should we just return early if it has been loaded? So that we won't have to trigger the setTimeout again and won't have to do more timeout cleanups.

Reply:
good idea, added.